.github/workflows/nightly_release.yml

```diff
@@ -0,0 +1,100 @@
+name: SalesforceCommerceCloud/pwa-kit/nightly_release
+on:
+  workflow_dispatch:
+  schedule:
+    # Run weekdays at 12am (PST) - cron uses UTC times
+    - cron: "0 8 * * 1-5"
+
+jobs:
+  create_nightly_release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get Timestamp
+        run: |-
+          echo "release_timestamp=$(date +'%Y%m%d%H%M%S')" >> "$GITHUB_ENV"
+
+      - name: Get monorepo version
+        run: |-
+          version=`jq -r ".version" package.json | cut -d "-" -f 1`
+          echo "monorepo_version_base=$version" >> "$GITHUB_ENV"
+          version_major=`echo "$version" | cut -d "." -f 1`
+          echo "monorepo_version_major=$version_major" >> "$GITHUB_ENV"
+
+      - name: Get retail-react-app version
+        run: |-
+          version=`jq -r ".version" packages/template-retail-react-app/package.json | cut -d "-" -f 1`
+          echo "retail_app_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Get commerce-sdk-react version
+        run: |-
+          version=`jq -r ".version" packages/commerce-sdk-react/package.json | cut -d "-" -f 1`
+          echo "commerce_sdk_react_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+
+      - name: Install Monorepo Dependencies
+        run: |-
+          # Install node dependencies
+          node ./scripts/gtime.js monorepo_install npm ci
+
+      - name: Create Release Branch
+        run: git switch -c release-nightly-${{ env.monorepo_version_major }}.x
+
+      - name: Pull latest changes from develop
+        run: git merge develop
+
+      - name: Bump version (monorepo)
+        run: |-
+          npm run bump-version -- "${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}"
+
+      - name: Bump version (retail-react-app)
+        run: |-
+          npm run bump-version:retail-react-app -- "${{ env.retail_app_version_base }}-nightly-${{ env.release_timestamp }}"
+
+      - name: Bump version (commerce-sdk-react)
+        run: |-
+          npm run bump-version:commerce-sdk-react -- "${{ env.commerce_sdk_react_version_base }}-nightly-${{ env.release_timestamp }}"
+
+      - name: Push branch to origin (Publish to NPM)
+        run: |-
+          git config --global user.name ${{ secrets.GIT_CONFIG_USERNAME }}
+          git config --global user.email ${{ secrets.GIT_CONFIG_EMAIL }}
+          git commit -am "Release ${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}"
+          git push --set-upstream origin release-nightly-${{ env.monorepo_version_major }}.x
+
+      - name: Create git tag
+        run: |-
+          git tag v${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}
```

Comment:
Totally inconsequential, but might be convenient to set `env.nightly_version = ${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}`, given how much it's used.